doc/book/examples/sources/patterns/facade.move

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
```

Comment:
Wondering if we should expand this into "type encapsulation" as a pattern by having an `option`/`vector` like pattern (where one module wraps a type and adds restrictions and/or extra functionality to it). I think this is more common, whereas I can't think of many cases where a facade at the function-only level makes sense.

Reply:
I temporarily removed the facade from everywhere. Need to study it better (also abstract classes like RegulatedCoin).